tests/unit/test_scmdataframe/test_scmdataframe.py

```diff
@@ -0,0 +1,1360 @@
+        pd.Index(
+            [
+                "model",
+                "scenario",
+                "region",
+                "variable",
+                "unit",
+                "climate_model",
+                "meta_int",
+            ]
+        ),
+    )
+
+
+def test_set_meta_as_str(test_scm_df):
+    test_scm_df.set_meta("testing", name="meta_str")
+
+    exp = pd.Series(
+        data=["testing", "testing", "testing"],
+        index=test_scm_df.meta.index,
+        name="meta_str",
+    )
+
+    obs = test_scm_df["meta_str"]
+    pd.testing.assert_series_equal(obs, exp)
+    pd.testing.assert_index_equal(
+        test_scm_df.meta.columns,
+        pd.Index(
+            [
+                "model",
+                "scenario",
+                "region",
+                "variable",
+                "unit",
+                "climate_model",
+                "meta_str",
+            ]
+        ),
+    )
+
+
+def test_set_meta_as_str_list(test_scm_df):
+    test_scm_df.set_meta(["testing", "testing2", "testing2"], name="category")
+    obs = test_scm_df.filter(category="testing")
+    assert obs["scenario"].unique() == "a_scenario"
+
+
+def test_set_meta_as_str_by_index(test_scm_df):
+    idx = pd.MultiIndex(
+        levels=[["a_iam"], ["a_scenario"]],
+        codes=[[0], [0]],
+        names=["model", "scenario"],
+    )
+
+    test_scm_df.set_meta("foo", "meta_str", idx)
+
+    obs = pd.Series(test_scm_df["meta_str"].values)
+    pd.testing.assert_series_equal(obs, pd.Series(["foo", "foo", None]))
+    pd.testing.assert_index_equal(
+        test_scm_df.meta.columns,
+        pd.Index(
+            [
+                "model",
+                "scenario",
+                "region",
+                "variable",
+                "unit",
+                "climate_model",
+                "meta_str",
+            ]
+        ),
+    )
+
+
+def test_filter_by_bool(test_scm_df):
+    test_scm_df.set_meta([True, False, False], name="exclude")
+    obs = test_scm_df.filter(exclude=True)
+    assert obs["scenario"].unique() == "a_scenario"
+
+
+def test_filter_by_int(test_scm_df):
+    test_scm_df.set_meta([1, 2, 3], name="test")
+    obs = test_scm_df.filter(test=[1])
+    assert obs["scenario"].unique() == "a_scenario"
+
+
+def test_rename_variable(test_scm_df):
+    mapping = {
+        "variable": {
+            "Primary Energy": "Primary Energy|Total",
+            "Primary Energy|Coal": "Primary Energy|Fossil",
+        }
+    }
+
+    obs = test_scm_df.rename(mapping)
+
+    exp = pd.Series(
+        ["Primary Energy|Total", "Primary Energy|Fossil", "Primary Energy|Total"]
+    )
+    pd.testing.assert_series_equal(
+        obs["variable"], exp, check_index_type=False, check_names=False
+    )
+
+
+def test_rename_index_fail(test_scm_df):
+    mapping = {"scenario": {"a_scenario": "a_scenario2"}}
+    pytest.raises(ValueError, test_scm_df.rename, mapping)
+
+
+@pytest.mark.skip
+def test_convert_unit():
```

Comment:
I think we want this to fail for this test case. We should be using unit registry internally and hence if you try to convert with units that aren't in the unit registry, things should just blow up. We should, of course, provide and test a method to convert units from say GtC to GtCO2 etc.